### PR TITLE
Windows: Rewrite WM_NCHITTEST & WM_NCCALCSIZE handling.

### LINKED ISSUE #109: ListView not easy press and hold the mouse to scroll in edge
image
在使用Listview不容易按住鼠标拖动滑块滚动, 会被判定为左右拉伸窗口. 
When using listview, it is not easy to hold down the mouse and drag the slider to scroll. It will be judged as stretching the window left and right

### PR DESCRIPTION
**Pull request by alexmercerind**
- Now resize hitbox lies outside the window frame (fixes #109 & #110).
- Use 0 MARGINS in `DwmExtendFrameIntoClientArea`.
- Removed `WM_CAPTION` from the parent `HWND`.
- No window border is visible on Windows 10 or 11.
- Reference: https://github.com/melak47/BorderlessWindow/issues/13.

**Windows 11**

78LbgdC6sK

**Windows 10**

harmonoid_BGzL58HNri

Other changes:

- I also merged Windows 7 support PR from @abutcher-gh.

### CODE REVIEW

**Comment by alexmercerind** [comment]
Hey @bitsdojo,

I know open source hasn't been much rewarding for all the hard-work. Unfortunately, I'm just a student so can't sponsor your work. Anyway, feel free to have a look/merge when you get time.

Best,
@alexmercerind

File: bitsdojo_window_linux/linux/bitsdojo_window_plugin.cpp [hunk 1]
```diff
@@ -111,8 +111,8 @@ FlBitsdojoWindowPlugin* bitsdojo_window_plugin_new(FlPluginRegistrar* registrar)
 
 void bitsdojo_window_plugin_register_with_registrar(FlPluginRegistrar* registrar) {
   FlBitsdojoWindowPlugin* plugin = bitsdojo_window_plugin_new(registrar);
-  FlView* view = fl_plugin_registrar_get_view(plugin->registrar);
-  enhanceFlutterView(GTK_WIDGET(view));
+  //  FlView* view = fl_plugin_registrar_get_view(plugin->registrar);
+  //  enhanceFlutterView(GTK_WIDGET(view));
```
Comment (alexmercerind) [inline bitsdojo_window_linux/linux/bitsdojo_window_plugin.cpp:115]: Sorry I had to disable this code for personal reason, it's not ideal to change window parameters unless asked intentionally. I don't know why was this necessary, when we already have a hook in Linux implementation.

**Comment by bitsdojo** [comment]
Hi @alexmercerind

Appreciate your message.

I've been offline in the past month and a half due to a medical issue but I will have access to my computer next week and I will take a look over the issues.

Bit

**Comment by ashutosh1211** [comment]
it still does not work on windows 7 SP1 64bit

image

**Comment by alexmercerind** [comment]
> it still does not work on windows 7 SP1 64bit

Thanks for the report. I will have a look & will require your feedback again.

**Comment by alexmercerind** [comment]
> it still does not work on windows 7 SP1 64bit
> 
> image

@ashutosh1211 I can't seem to find any unsafe calls. Make sure you have the `bitsdojo_window_windows` registered like this (with dependency overrides) :
https://github.com/harmonoid/harmonoid/blob/b74a9d0ba8349334b990f023f44bab206069307f/pubspec.yaml#L70-L74
https://github.com/harmonoid/harmonoid/blob/b74a9d0ba8349334b990f023f44bab206069307f/pubspec.yaml#L83-L94

**Comment by ashutosh1211** [comment]
> 

Thank you. It launches on windows 7 now without any error.

image

**Comment by jibbe** [comment]
@alexmercerind the window border seems to wider now, is this intended?

**Comment by alexmercerind** [comment]
> @alexmercerind the window border seems to wider now, is this intended?

On Windows 7 & 8.x it is. And there's no workaround if you want the "re-size area" to stay outside the window frame. 

See Google Chrome on Windows 7 or 8.x, it will also have this border.
